### PR TITLE
Facebook App Events fixes

### PR DESCRIPTION
**Pull request by dhawal1248**

### CODE REVIEW

File: v0/destinations/fb/transform.js [hunk 1]
```diff
@@ -97,43 +115,75 @@ function sanityCheckPayloadForTypesAndModifications(updatedEvent) {
...
-          updatedEvent.custom_events[0][fbEventPath][length] =
-            intendValue || "";
-
-          length++;
-          count--;
+  const { properties } = message;
+  if (properties) {
+    if (properties.revenue && !properties.currency) {
+      throw new Error(
+        "If properties.revenue is present, properties.currency is required."
```
Comment (arnab-p) [inline v0/destinations/fb/transform.js:144]: Can we set to something default like `USD`? Does FB do it?

**Review by arnab-p — CHANGES_REQUESTED** [review]
There are bunch of TODO comments in the code. Should we remove it or kept for reference for future efforts?

File: v0/destinations/fb/transform.js (hunk 1 above)
Comment (sayan-mitra) [reply v0/destinations/fb/transform.js:144]: I think better to have explicit ?
https://developers.facebook.com/docs/marketing-api/app-event-api (fb_currency)

**Comment by sayan-mitra** [comment]
> There are bunch of TODO comments in the code. Should we remove it or kept for reference for future efforts?

yeah

**Review by arnab-p — APPROVED** [review]

**Pull request merged**